### ISSUE TITLE
tf.Estimator starts with GPU and switches to CPU

### ISSUE THREAD

**Issue by arozans** [tensorflow/tensorflow#31549]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 16.04.4 LTS (Xenial Xerus)
- TensorFlow installed from (source or binary): from pip
- TensorFlow version (use command below): 1.14, but earlier also for 1.11
- Python version: 3.6
- CUDA/cuDNN version: 10.0.130 / 7.6.2
- GPU model and memory: 1080Ti 11GB

Few days ago I've asked a SO question: https://stackoverflow.com/questions/57097880/tensorflow-estimator-gradually-decreasing-gpu-utilization - sadly without a response, so I'm trying here, not sure if this is a bug.

Copied from the question:
I'm training tf.Estimator model (5025056 trainable parameters):

    # simplified code
    model = create_custom_model() #preparing my model here 
    tf.reset_default_graph()
    estimator = tf.estimator.Estimator(model_fn=model.get_model_fn())
    evaluation_hook = tf.contrib.estimator.InMemoryEvaluatorHook(
        estimator=estimator,
        input_fn=lambda: model.eval_input_fn()
    )
    estimator.train(
        input_fn=lambda: model.train_input_fn(),
        hooks=[evaluation_hook]
    )
And dataset is prepared here:

    def train_input_fn():
        dataset = tf.data.TFRecordDataset("some_filename")
        dataset = dataset.shuffle(3000)
        dataset = dataset.repeat()
        dataset = dataset.batch(384)
        dataset = dataset.prefetch(1)
        return dataset

My dataset consists of images, 9000 samples, stored in TFRecord (163M). 
GPU is `GeForce GTX 1080 Ti` and CPU is `i5-6600 CPU @ 3.30GHz`.
During the training, at first, everything looks fine - `htop` shows that each core is working approximately the same way (mostly jumping between `0` - `50%` utilization), and gpu stats (shown using `vidia-smi --query-gpu=timestamp,pstate,temperature.gpu,utilization.gpu,utilization.memory,memory.total,memory.free,memory.used --format=csv -l 1`) indicate that card is working on `100%` of its power - temperatures around `85°C`, `100%` utilization of gpu and memory, also most of the memory is used(10GB).
Up to step `4000` (of `10000`) tensorflow prints `global_step/sec` with values ranging from `1.0` to `1.1`. `15GB` of available `16GB` of RAM is used and no swap.

After that step, that time `global_step/sec` is lower and lower. For step `5000` it is `0.617275`.
Aaround step 2000 it is only `0.0938672` and decreasing (down to `0.0368357` at the end of the training). During this process `nvidia-smi` shows that `utilization.gpu` and `utilization.memory` is `0%` more and more frequently (despite the fact that `memory.used` shows same amount whole time, that is `10GB`), and CPU is working at a `100%` of a single core. RAM is at the same level as when training started, and no swapping occurs. Periodically, mayby once for `20` steps, GPU utilization is slightly higher, but quickly returns to `0`. 

It looks like after some epochs tensorflow trains on CPU instead of GPU? What can be possible cause of that?

**Comment by ravikyram** [comment]
Looks like the code is incomplete.Can you please provide full code snippet to reproduce it on our environment.Thanks!

**Comment by arozans** [comment]
Full project is available in my repo here: https://github.com/arozans/idenface/tree/IF-NONE_GitHub_GPU_issue.

Entry point is file `training.py` (training with multiple models in a loop). From my observations -  after first (sometimes second) model finishes training, GPU often goes into low performance P8 state. Attaching also log for this training, grepping with " sec)" show diminishing training speed.
[t_e_0821t0100.log](https://github.com/tensorflow/tensorflow/files/3519486/t_e_0821t0100.log)

**Comment by arozans** [comment]
Hey, any news on that one? Any suggestion where to look? I'm attaching one more log, this time with corresponding gpu (from `nvidia-smi`) and cpu (from `ps`). (Usage of cpu with time is also decreasing, because after some time only one core is working on 100%)
[cnn_training_0921t2132.log](https://github.com/tensorflow/tensorflow/files/3639879/cnn_training_0921t2132.log)
[cpu.log](https://github.com/tensorflow/tensorflow/files/3639880/cpu.log)
[gpu.log](https://github.com/tensorflow/tensorflow/files/3639881/gpu.log)

**Comment by aaroey** [comment]
I'm not familiar with Estimator. @tanzhenyu could you help to find someone to take a look? Thanks.

**Comment by sanjoy** [comment]
Hi @arozans,

I would suggest poking at your training job with the [TensorBoard profiler](https://www.tensorflow.org/tensorboard/tensorboard_profiling_keras) to see if you can spot any red flags.  You can using the [profiler APIs](https://www.tensorflow.org/tensorboard/tensorboard_profiling_keras#profiler_apis) to profile the program for a few steps only after step `4000` (since that is when the problem starts).

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.
